server/src/test-fast/java/com/thoughtworks/go/server/service/plugins/processor/serverinfo/v2/MessageHandlerForServerInfoRequestProcessor2_0Test.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.service.plugins.processor.serverinfo.v2;
+
+import com.thoughtworks.go.CurrentGoCDVersion;
+import com.thoughtworks.go.config.ServerConfig;
+import org.junit.Test;
+
+import static java.lang.String.format;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class MessageHandlerForServerInfoRequestProcessor2_0Test {
+    @Test
+    public void shouldSerializeServerConfigToJSON() {
+        ServerConfig serverConfig = new ServerConfig();
+        serverConfig.ensureServerIdExists();
+        serverConfig.setSecureSiteUrl("https://example.com:8154/go");
+        serverConfig.setSiteUrl("http://example.com:8153/go");
+        CurrentGoCDVersion goCDVersion = CurrentGoCDVersion.getInstance();
+
+        MessageHandlerForServerInfoRequestProcessor2_0 processor = new MessageHandlerForServerInfoRequestProcessor2_0();
+
+        assertThat(processor.serverInfoToJSON(serverConfig),
```

Comment:
Use `assertThatJson` instead of matching JSON strings. There's no guarantee of ordering of the different keys in JSON. This is likely to cause flaky tests.

Reply:
@ketan assertThatJson expects a json object. but the return type here is String. We could otherwise use `JSONAssert.assertEquals`.